src/Adapter/Image/ImageRetriever.php

```diff
@@ -205,7 +207,7 @@ public function getImage($object, $id_image)
             $url = $this->link->$getImageURL(
                 isset($object->link_rewrite) ? $object->link_rewrite : $object->name,
                 $id_image,
-                $image_type['name']
+                $image_type['name'] . ($generateHighDpiImages ? self::HIGH_RES_URL_SUFFIX : '')
```

Comment:
```suggestion
                $image_type['name'] . ($generateHighDpiImages ? static::HIGH_RES_URL_SUFFIX : '')
```

Reply:
I don't think you can do that here, the `generateHighDpiImages` comes from a configuration if you want to enable high resolution images That's fine and it's related to your shop's config
However it should only impact the image generation no the image retrieval and force the URL for everyone

The image quality depends on the FO and especially the user agent, it should be handled in the front theme not here (as you said it's mostly interesting for mobile devices with retina screen, which is not the case for all devices)